compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -462,7 +463,7 @@ convertInterfaces env binds = interfaceDefs
         let precond = fromMaybe (error $ "Missing precondition for interface " <> show intName)
                         $ (MS.lookup intName $ envInterfaceBinds env) >>= ibEnsure
         withRange intLocation $ do
-            intRequires <- fmap S.fromList $ mapM (convertInterfaceTyCon env) $
+            intRequires <- fmap S.fromList $ mapM (convertInterfaceTyCon env $ unhandled "interface type") $
```

Comment:
Ah, you uncovered a related issue. This corresponds to 

```haskell
interface I requires Bool
```

We should give a similar error message here, something like `cannot require 'Bool' because it is not an interface`

Reply:
After this, could you please add a couple of test files under [`compiler/damlc/tests/daml-test-files`](https://github.com/digital-asset/daml/tree/main/compiler/damlc/tests/daml-test-files)? These should be plain `.daml` files that exhibit the errors (one for implementing a non-interface, another for requiring one), except they should have a special comment of the form

```haskell
-- @ERROR range=8:1-9:17; <error message>
```

the testsuite will then check that `damlc` reports the error at the specified range. 

You can run all the tests under `compiler/damlc/tests/daml-test-files` with

```bash
bazel test //compiler/damlc/tests:integration-v1dev
```

or you can filter them based on their filename with

```bash
bazel test //compiler/damlc/tests:integration-v1dev --test_arg -p --test_arg "MyTestFile"
```

We don't have many rules for naming these files, but we're trying to at least prefix the interface-related ones with `Interface`, so these would be something like `InterfaceImplementNonInterface.daml` and `InterfaceRequireNonInterface.daml`

Reply:
Finally got some amount of source-mapping working :|
Tests are in and passing
Handlers simplified/RankNTypes removed (https://github.com/digital-asset/ghc/pull/126)

Reply:
thank you!